### ISSUE TITLE
[Bug] After setUserMatrix, Actor become pure black.

### ISSUE THREAD

**Issue by neo15201788751** [Kitware/vtk-js#3148]
### Bug description

Loading polydata with vtkSTLReader from stl file, then create an vtkActor, setUserMatrix,
 the actor looks strange, it become pure black.

### Steps to reproduce

Please see, [demo](https://codesandbox.io/p/sandbox/hello-world-cone-forked-r4shvf)

### Expected Behavior

Just play right as without setUserMatrix.

### Environment

- vtk.js version:
- Browsers:
- OS:

**Comment by finetjul** [comment]
This is likely not a bug, it is because your user matrix "flips" the point normals. It is therefore assumed that the normals are pointing towards the center of the object instead of pointing outward. The lighting mechanism needs to have the dot product between the normals and the light direction is negative so that it lit correctly.

**Comment by neo15201788751** [comment]
How to update the point normals to make it towards the center? I think, this should be done in method: setUserMatrix.

**Comment by finetjul** [comment]
No, it shouldn't be done in UserMatrix. The flip might be intentional.

You can look at `vtkPolyDataNormals`

**Comment by neo15201788751** [comment]
Where to see the flipped normals? I do not know when and where to update the normals.

**Comment by finetjul** [comment]
> Where to see the flipped normals?

You can't see them (because the user matrix is applied at render time), but the blackness is explained by flipepd normals.

> I do not know when and where to update the normals.

You can set the output of the vtkPolyDataNormals filter as the input of the mapper.

**Comment by finetjul** [comment]
> Thanks. I need to know if the matrix causes normals flipping, then use vtkPolyDataNormals to fix it, or do nothing.
> Not all matrix will cause flipping. How to know matrix will cause flipping or not?

If an axis from the matrix is flipped
